app/api/server/v1/roles.js

```diff
@@ -104,3 +123,53 @@ API.v1.addRoute('roles.getUsersInRole', { authRequired: true }, {
 		return API.v1.success({ users: users.fetch(), total: users.count() });
 	},
 });
+
+API.v1.addRoute('roles.update', { authRequired: true }, {
+	post() {
+		check(this.bodyParams, {
+			roleId: String,
+			name: Match.Maybe(String),
+			scope: Match.Maybe(String),
+			description: Match.Maybe(String),
+		});
+
+		const roleData = {
+			roleId: this.bodyParams.roleId,
+			name: this.bodyParams.name,
+			scope: this.bodyParams.scope,
+			description: this.bodyParams.description,
+		};
+
+		const role = Roles.findOneByIdOrName(roleData.roleId);
+
+		if (!role) {
+			throw new Meteor.Error('error-invalid-roleId', 'This role does not exist');
+		}
+
```

Comment:
the update should also throw an error if the user is trying to modify a protected role.